.github/workflows/build.yml

```
@@ -2,28 +2,25 @@ name: build
 
 on:
   push:
-    branches: 
-      - main
+    branches: [ main ]
   pull_request:
-    branches:
-      - "*"
+    branches: [ main ]
 
 jobs:
   Build:
     strategy:
-      fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest]
     runs-on: ${{ matrix.os }}
-    defaults:
-      run:
-        shell: bash
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up JDK 8
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Configure Java
         uses: actions/setup-java@v2
         with:
-          java-version: "8"
-          distribution: "adopt"
+          java-version: 8
```

I think it is safer to set it as `string`:

```suggestion
          java-version: '8'
```

I remember that in the past I had some problems that such thing was interpreted as a number

Moreover this is what is suggested by docs: https://github.com/actions/setup-java

---

Thank you, but you commented outdated code 😛